multihost_test/Bugzillas/test_subid_ranges_configurable.py

```diff
@@ -0,0 +1,294 @@
+        multihost.client[0].run_command("gcc /tmp/list_subid_ranges.c -lsubid -o "
+                                        "/tmp/list_subid_ranges")
+        cmd = multihost.client[0].run_command("cd /tmp/; ./list_subid_ranges "
+                                              "local_anuj")
+        assert 'local_anuj 165536 65536' in cmd.stdout_text
+        cmd = multihost.client[0].run_command("cd /tmp/; ./list_subid_ranges -g "
+                                              "local_anuj")
+        assert 'local_anuj 165536 65536' in cmd.stdout_text
+
+    def test_0006_compiler_unexisting(self, multihost,
+                                      create_bkp_for_1859252,
+                                      localuser_for_1859252):
+        """
+        :Title: support pluggable data sources for
+        subid ranges configurable via /etc/nsswitch.conf
+        :id: b7ca349e-00c1-11ec-b959-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1859252
+        :steps:
+            1. Define user range in /etc/subuid
+            2. Define group range in /etc/subgid
+            3. Create a process for user
+            4. Check list_subid_ranges command is working
+            with subid: unexisting in /etc/nsswitch.conf
+            5. Check list_subid_ranges -g command is working
+            with subid: unexisting in /etc/nsswitch.conf
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+        """
+        exceute_cmd(multihost, "yum --enablerepo=rhel-CRB install shadow-utils*")
+        multihost.client[0].run_command("yum install -y shadow-utils-subid-*")
+        multihost.client[0].run_command("yum install -y gcc")
+        multihost.client[0].run_command("echo 'local_anuj:165536:65536'"
+                                        " > /etc/subuid")
+        multihost.client[0].run_command("echo 'local_anuj:165536:65536'"
+                                        " > /etc/subgid")
+        multihost.client[0].run_command("echo 'subid: unexisting' "
+                                        ">> /etc/nsswitch.conf")
+        multihost.client[0].transport.put_file(os.getcwd()+'/data/list_subid_ranges.c',
+                                               '/tmp/list_subid_ranges.c')
+        multihost.client[0].run_command("gcc /tmp/list_subid_ranges.c -lsubid -o "
+                                        "/tmp/list_subid_ranges")
+        cmd = multihost.client[0].run_command("cd /tmp/; ./list_subid_ranges "
+                                              "local_anuj")
+        assert 'local_anuj 165536 65536' in cmd.stdout_text
+        cmd = multihost.client[0].run_command("cd /tmp/; ./list_subid_ranges -g "
+                                              "local_anuj")
+        assert 'local_anuj 165536 65536' in cmd.stdout_text
```

Comment:
I'm missing a test that plays with `SUB_UID_COUNT` and `SUB_GID_COUNT` from `/etc/login.defs`. These configurations define the value that will be used to generate the subid range.

I'd expect to see at least one test that changes the value of `SUB_UID_COUNT` to X and `SUB_GID_COUNT` to Y, add a new user and use `list_subid_ranges`to check that the uid range is equal to X and the gid range is equal to Y.

Reply:
that test will be added later on top of the same file . for now i would like see these tests merged . After that one by one we can put lots of others test on same file .